integration/ig/k8s/enrichment_pod_label_test.go

```diff
@@ -0,0 +1,218 @@
+					c.Runtime.ContainerImageName = ""
+				}
+			}
+
+			ExpectEntriesInArrayToMatch(t, output, normalize, expectedContainer)
+		},
+	}
+
+	RunTestSteps([]*Command{listContainersCmd}, t, WithCbBeforeCleanup(PrintLogsFn(ns)))
+}
+
+func TestEnrichmentPodLabelNewPod(t *testing.T) {
+	t.Parallel()
+
+	cn := "test-enrichment-pod-label-new-pod"
+	pod := cn
+	ns := GenerateTestNamespaceName(pod)
+
+	listContainersCmd := &Command{
+		Name:         "RunWatchContainers",
+		Cmd:          fmt.Sprintf("ig list-containers -o json --runtimes=%s --use-cri --watch", *containerRuntime),
+		StartAndStop: true,
+		ValidateOutput: func(t *testing.T, output string) {
+			isDockerRuntime := *containerRuntime == ContainerRuntimeDocker
+			expectedEvent := &containercollection.PubSubEvent{
+				Type: containercollection.EventTypeAddContainer,
+				Container: &containercollection.Container{
+					K8s: containercollection.K8sMetadata{
+						BasicK8sMetadata: types.BasicK8sMetadata{
+							ContainerName: cn,
+							PodName:       pod,
+							Namespace:     ns,
+							PodLabels: map[string]string{
+								"io.kubernetes.container.name": "POD",
+								"io.kubernetes.pod.name":       pod,
+								"io.kubernetes.pod.namespace":  ns,
+								"io.kubernetes.pod.uid":        "",
+								"run":                          cn,
+							},
+						},
+					},
+					Runtime: containercollection.RuntimeMetadata{
+						BasicRuntimeMetadata: types.BasicRuntimeMetadata{
+							RuntimeName:        types.String2RuntimeName(*containerRuntime),
+							ContainerName:      cn,
+							ContainerImageName: "docker.io/library/busybox:latest",
+						},
+					},
+				},
+			}
+
+			// Docker can provide different values for ContainerImageName. See `getContainerImageNamefromImage`
+			if isDockerRuntime {
+				expectedEvent.Container.Runtime.ContainerImageName = ""
+			}
+
+			normalize := func(e *containercollection.PubSubEvent) {
+				e.Container.Pid = 0
+				e.Container.OciConfig = nil
+				e.Container.Bundle = ""
+				e.Container.Mntns = 0
+				e.Container.Netns = 0
+				e.Container.CgroupPath = ""
+				e.Container.CgroupID = 0
+				e.Container.CgroupV1 = ""
+				e.Container.CgroupV2 = ""
+				e.Timestamp = ""
+
+				e.Container.SandboxId = ""
+				e.Container.K8s.PodLabels["io.kubernetes.pod.uid"] = ""
+				e.Container.K8s.PodUID = ""
+				e.Container.Runtime.ContainerID = ""
+				e.Container.Runtime.ContainerImageDigest = ""
+
+				// Docker and CRI-O use a custom container name composed, among
+				// other things, by the pod UID. We don't know the pod UID in
+				// advance, so we can't match the expected container name.
+				// TODO: Create a test for this once we support filtering by k8s
+				// container name. See
+				// https://github.com/inspektor-gadget/inspektor-gadget/issues/1403.
+				if e.Container.Runtime.RuntimeName == ContainerRuntimeDocker ||
+					e.Container.Runtime.RuntimeName == ContainerRuntimeCRIO {
+					e.Container.Runtime.ContainerName = cn
```

Comment:
For the above test we can query the pod UID and then construct the container name from that. In this test `TestEnrichmentPodLabelNewPod` we start the `test-pod` after we start `ig`. So we don't have knowledge about the pod before and don't know what pod UID it will have